workflow/controller/operator.go

```diff
@@ -3832,6 +3843,44 @@ func (woc *wfOperationCtx) retryStrategy(tmpl *wfv1.Template) *wfv1.RetryStrateg
 	return woc.execWf.Spec.RetryStrategy
 }
 
+func (woc *wfOperationCtx) shouldRetry() bool {
+	retryConfig := woc.execWf.Spec.Retry
+	if retryConfig == nil {
+		return false
+	}
+	if woc.IsRetried() {
+		if woc.controller.podCleanupQueue.Len() == 0 {
```

Comment:
what is the purpose of this line here?

Reply:
I just want to confirm that all the pod has been cleaned up. If exists is not reported when creating a new pod at this time, this should be removed.

Reply:
Do you mean something like
make sure all pods deleted before process retry workflow

Reply:
Yes, this is what I want.

Reply:
I think maybe we should check nodestatus instead.
Other workflows would queue up the podCleanupQueue causing issue